app/models/domain_block.rb

```diff
@@ -12,11 +12,12 @@
 #
 
 class DomainBlock < ApplicationRecord
-  enum severity: [:silence, :suspend]
+  enum severity: [:silence, :suspend, :media_only]
```

Comment:
No, I think it should be called accordingly (Transparent or No-op). Solves you a lot of trouble, no more need for the validation, even hiding the boolean with JS could be omitted